nuxeo-features/nuxeo-annotation/nuxeo-annotation-rest-api/src/test/java/org/nuxeo/ecm/restapi/server/AnnotationTest.java

```diff
@@ -0,0 +1,230 @@
+        DocumentModel file = session.createDocumentModel("/", "testDoc", "File");
+        file = session.createDocument(file);
+
+        fetchInvalidations();
+        String xpath = "file:content";
+
+        Annotation annotation = new AnnotationImpl();
+        annotation.setColor("000000");
+        annotation.setLastModifier("bob");
+        annotation.setPosition("0,0,0,0");
+
+        String jsonAnnotation = MarshallerHelper.objectToJson(annotation, CtxBuilder.get());
+
+        try (CloseableClientResponse response = getResponse(RequestType.POST,
+                "id/" + file.getId() + "/@annotation/" + xpath, jsonAnnotation)) {
+            assertEquals(Response.Status.CREATED.getStatusCode(), response.getStatus());
+            JsonNode node = mapper.readTree(response.getEntityInputStream());
+            assertEquals(file.getId(), node.get("documentId").textValue());
+            assertEquals(xpath, node.get("xpath").textValue());
+            assertEquals("000000", node.get("color").textValue());
+            assertEquals("bob", node.get("lastModifier").textValue());
+            assertEquals("0,0,0,0", node.get("position").textValue());
+        }
+    }
+
+    @Test
+    public void testGetAnnotation() throws IOException {
+        DocumentModel file = session.createDocumentModel("/", "testDoc", "File");
+        file = session.createDocument(file);
+        String xpath = "files:files/0/file";
+
+        Annotation annotation = new AnnotationImpl();
+        Calendar now = Calendar.getInstance();
+        annotation.setCreationDate(now);
+        annotation.setPage(42L);
+        annotation = annotationService.createAnnotation(session, file.getId(), xpath, annotation);
+        fetchInvalidations();
+
+        JsonNode node = getResponseAsJson(RequestType.GET,
+                "id/" + file.getId() + "/@annotation/" + xpath + "/" + annotation.getId());
+
+        assertEquals(AnnotationJsonWriter.ENTITY_TYPE, node.get("entity-type").asText());
+        assertEquals(annotation.getId(), node.get("id").textValue());
+        assertEquals(42L, node.get("page").longValue());
+        assertEquals(file.getId(), node.get("documentId").textValue());
+        assertEquals(xpath, node.get("xpath").textValue());
+        assertEquals(now.toInstant().toString(), node.get("creationDate").textValue());
+    }
+
+    @Test
+    public void testUpdateAnnotation() throws IOException {
+        DocumentModel file = session.createDocumentModel("/", "testDoc", "File");
+        file = session.createDocument(file);
+
+        fetchInvalidations();
+        String xpath = "file:content";
+
+        Annotation annotation = new AnnotationImpl();
+        annotation.setOpacity(0.5);
+        annotation.setName("testAnnotation");
+        annotation = annotationService.createAnnotation(session, file.getId(), xpath, annotation);
+
+        fetchInvalidations();
+
+        annotation.setName("testUpdatedAnnotation");
```

Comment:
Could create a n ew `Annotation` here ? As the code rely on a `PUT`, we should be able to just send the name and/or security and updates will be down against the persisted objects.
Also this will allow you to check that we don't need to put id in both path and body object.